core/src/main/java/com/novoda/merlin/service/MerlinService.java

```diff
@@ -82,6 +82,7 @@ protected ComponentName connectivityReceiverComponent() {
 
     public void setHostname(String hostname, ResponseCodeValidator validator) {
         hostPinger = HostPinger.withCustomEndpointAndValidation(this, hostname, validator);
+        currentNetworkStatusRetriever = new CurrentNetworkStatusRetriever(MerlinsBeard.from(this.getApplicationContext()), hostPinger);
```

Comment:
You don't need the `this`, right? Besides, this smells a bit of unexpected side effect (but I see the point). Maybe extract a setter for the retriever?